### PR TITLE
Fixes and reenables rubocop Lint/ShadowingOuterLocalVariable.

### PR DESCRIPTION
**Pull request by ashercodeorg**
All changes manually done, as the cop does not support `--auto-correct`.

### CODE REVIEW

**Comment by ashercodeorg** [comment]
Note that, in some cases, I eliminated usages of `tap`. Feel welcome to push back on these changes, I just (a) didn't see any obvious renaming of the shadowed variable and (b) didn't see the advantage of using `tap`.

**Comment by joshlory** [comment]
This change looks reasonable to me.  Maybe worth having @wjordan take a look at the `.tap()` changes, he's a fan of that pattern.

File: bin/cron/geocode_hoc_activity [hunk 1]
```diff
@@ -7,31 +7,30 @@ BATCH_SIZE = 4000
...
-    THREAD_COUNT.times do |segment|
-      threads << Thread.new do
-        counter_key = "hoc_activity.last_id-#{segment}"
-        at = Properties.get(counter_key).to_i
-        DB.fetch(
-          "SELECT id,started_ip,pixel_started_ip,pixel_finished_ip,finished_ip FROM hoc_activity WHERE id > #{at} AND id < #{at + BATCH_SIZE} ORDER BY id ASC"
-        ).each do |row|
-          next if (row[:id] % THREAD_COUNT) != segment
+  threads = []
+  THREAD_COUNT.times do |segment|
```
Comment (mehalshah) [inline bin/cron/geocode_hoc_activity:11]: Sure this does what you want? This is going to return 2, not the thread array

Comment (ashercodeorg) [inline bin/cron/geocode_hoc_activity:11]: Good catch. Adding `return threads` to method end.

Comment (aoby) [inline bin/cron/geocode_hoc_activity:11]: Looks like in this case the return value isn't important and isn't used. We just want to execute the block `THREAD_COUNT` times. @ashercodeorg please confirm.

Comment (ashercodeorg) [inline bin/cron/geocode_hoc_activity:11]: Actually, never mind, keeping as-is. The return isn't used, nor is it likely to ever be used.

**Comment by aoby** [comment]
LGTM

**Comment by mehalshah** [comment]
LGTM

**Pull request merged**